src/main/java/com/aws/greengrass/secretmanager/FileSecretDao.java

```diff
@@ -45,26 +47,59 @@ public FileSecretDao(KernelClient kernelClient) throws SecretManagerException {
 
     /**
      * Retrieve secret document from underlying file store.
+     *
      * @return {@link SecretDocument} containing all persisted secrets
      * @throws SecretManagerException when there is any issue reading the store.
      */
     public synchronized SecretDocument getAll() throws SecretManagerException {
+        SecretDocument secretDocFromServiceTopic = getSecretDocFromServiceTopic();
+        SecretDocument secretDocFromRuntimeTopic = getSecretDocFromRuntimeTopic();
+        ArrayList<AWSSecretResponse> allSecrets = new ArrayList<>();
+
+        if (secretDocFromServiceTopic != null) {
+            allSecrets.addAll(secretDocFromServiceTopic.getSecrets());
+        }
+
+        if (secretDocFromRuntimeTopic != null) {
+            allSecrets.addAll(secretDocFromRuntimeTopic.getSecrets());
+        }
+        if (Utils.isEmpty(allSecrets)) {
+            throw new NoSecretFoundException("No secrets found in the file");
+        }
```

Comment:
null and empty aren't the same thing, so this isn't really doing the same as the old code.

However, I don't think that the old code is actually useful by throwing the no secret exception. I think you can just remove this.